src/biom.cpp

```diff
@@ -77,30 +110,65 @@ biom::biom(std::string filename) {
                 sizeof(unsigned int) * n_obs, __FILE__, __LINE__);
         exit(EXIT_FAILURE);
     }
+}
+
+biom::biom() : has_hdf5_backing(false) { 
+    n_obs = 0;
+    malloc_resident(0);
+}
+
+biom::biom(char** obs_ids_in,
+           char** samp_ids_in,
+           const int32_t* indices,
+           const int32_t* indptr,
+           const double* data,
+           const int n_obs,
+           const int n_samples,
+           const int nnz) : has_hdf5_backing(false) {
+    this->nnz = nnz;
+    this->n_samples = n_samples;
+    this->n_obs = n_obs;
+
+    sample_ids = std::vector<std::string>();
+    sample_ids.reserve(n_samples);
+    obs_ids = std::vector<std::string>();
+    obs_ids.reserve(n_obs);
+
+    for(int i = 0; i < n_obs; i++) {
```

Comment:
Performance note:
If n_objs is large, we may want to create the various vector in parallel.